helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1360,8 +1390,20 @@ private boolean updateControllerState(NotificationContext changeContext, PauseSi
         String uid = UUID.randomUUID().toString().substring(0, 8);
         ClusterEvent event = new ClusterEvent(_clusterName, ClusterEventType.Resume,
             String.format("%s_%s", uid, Pipeline.Type.DEFAULT.name()));
+
+        HelixManager manager = changeContext.getManager();
+        Optional<String> leaderSession = manager.getSessionIdIfLead();
```

Comment:
Also using of manager.getSessionIdIfLead here could be confusing. We used getSessionId in other places, what is the difference?  Where we should use getSessionIdIfLead and where we should use getSessionId?

Reply:
`manager.getSessionIdIfLead` checks whether the cluster manager is leader and returns the session ID associated to the connection of cluster data store, if and only if it is leader. It combines `isLeader()` and `getSessionId()` in one method to ensure the session id returning is what we check leadership in isLeader().

Without it, we may do this:
```
if (isLeader()) {
  sessionId = manager.getSessionId();
}
```
The problem is race condition. sessionId may change after isLeader().

So we need `manager.getSessionIdIfLead` to check leadership and get sessionId at the same time.